### DISCUSSION TITLE
non cooperation while generating packages for cpp via graphics r ,d package groups

### DISCUSSION THREAD

**Discussion by MaheshNaikM** [graphics32/graphics32#263] [General]
package creation fails for the d part of gr32_groupproj. The R part translates properly. I wanted to translate to use via a c++ program.I was translating properly till  the 104 packages.  fails for the d270 package creation with delphi 10.4.Please help.I want to lift the packages to rad studio cpp 11.0

i can remove some trivial errors like rrequires instead of requires....,a missing end.
but the undefined error for TResourceType persists though somewhere it is redefined as PChar...
can i be helped for editing changes in the package generation for cpp usable paksc

**Comment by andersmelander** [comment]
I'm unsure what it is you are saying - or asking about, but I'll give it a shot anyway.

- Are you saying that there's a misspelled `requires` somewhere? If so, where?
- There's a missing `end`? Where?
- I haven't been able to detect any problems with regard to CR/LF. It might be a problem at your end with your Git client configuration.
- With regard to `TResourceType`: In `gr32.pas` try replacing:
  ```
    TResourceType = PChar;
  ```
  with
  ```
    (*$HPPEMIT 'typedef LPWSTR TResourceType;'*)
    TResourceType = PChar;
  ```
  Basically, `TResourceType` should be an alias for the return type of the `MakeIntResource` macro.

**Reply by MaheshNaikM** [reply]
When i use  Delphi to create two bpl files  for the package I get those funny errors of a repeated rr , also . for end. I edit them and the packages get created. I am using 10.4.2  and trying to port to  community edition  11 of cbuilder .

     Surprisingly i dont get the rrequires and missing end.   when i now use the package generator.?. 
I will try your edit  later.
I had managed to compile by deleting some code from the source(Graphics32) as i do not require the image part for polygon display!.
Thank you

**Reply by MaheshNaikM** [reply]
--description:"Graphics32 Design Time Package" -NBd:\pot -NHd:\pot -NO.\Win32\Debug  GR32_D.dpk   
[dcc32 Error] GR32_Dsgn_ColorPicker.pas(128): E2003 Undeclared identifier: 'Execute'
[dcc32 Fatal Error] GR32_Dsgn_Color.pas(1040): F2063 Could not compile used unit 'GR32_Dsgn_ColorPicker.pas'
Failed
Elapsed time: 00:00:00.8
GR32_Dsgn_ColorPicker.pas(1): Line endings are LF, but RAD Studio requires CRLF. Consider converting or check your source control settings.

**Reply by MaheshNaikM** [reply]
I did your edit ..it worked..but  some other errors propped up. are some files not having a crlf ?i have just showed one error above.
I have errors in GR32_DSGN_Bitmap also

**Comment by andersmelander** [comment]
> `[dcc32 Error] GR32_Dsgn_ColorPicker.pas(128): E2003 Undeclared identifier: 'Execute'`

The errors make no sense. As you can see in the source of `GR32_Dsgn_ColorPicker.pas` the `Execute` method is declared in line 105.
I think you just need to fix your line-ending problem.
